md-dao/src/main/java/com/wipro/ats/bdre/md/dao/JobDAO.java

```diff
@@ -239,7 +244,7 @@
             Integer notNullTargetBatches = 0;
             for (Object subProcessObject : listOfSubProcessCriteria.list()) {
                 Process subProcess = (Process) subProcessObject;
-                Criteria notNullTargetBatchCriteria = session.createCriteria(BatchConsumpQueue.class).add(Restrictions.isNotNull("batchByTargetBatchId")).add(Restrictions.eq("process", subProcess));
+                Criteria notNullTargetBatchCriteria = session.createCriteria(BatchConsumpQueue.class).add(Restrictions.isNotNull(BATCHBYTARGETBATCHID)).add(Restrictions.eq(PROCESS, subProcess));
                 if (notNullTargetBatchCriteria.list().size() > 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1155)